Services/Component/classes/Setup/class.ilComponentActivatePluginsObjective.php

```diff
@@ -0,0 +1,150 @@
+<?php
+/* Copyright (c) 2020 Daniel Weise <daniel.weise@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+declare(strict_types=1);
+
+use ILIAS\Setup;
+use ILIAS\DI;
+
+class ilComponentActivatePluginsObjective implements Setup\Objective
+{
+    /**
+     * @var string
+     */
+    protected $plugin_name;
+
+    public function __construct(string $plugin_name)
+    {
+        $this->plugin_name = $plugin_name;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getHash() : string
+    {
+        return hash("sha256", self::class . $this->plugin_name);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getLabel() : string
+    {
+        return "Activate plugin $this->plugin_name.";
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function isNotable() : bool
+    {
+        return true;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getPreconditions(Setup\Environment $environment) : array
+    {
+        $setup_config = $environment->getConfigFor('common');
+        $db_config = $environment->getConfigFor('database');
+
+        return [
+            new \ilIniFilesPopulatedObjective($setup_config),
+            new \ilDatabasePopulatedObjective($db_config),
+            new \ilComponentPluginAdminInitObjective()
+        ];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function achieve(Setup\Environment $environment) : Setup\Environment
+    {
+        $ORIG_DIC = $this->initEnvironment($environment);
+
+        $plugin = $GLOBALS["DIC"]["ilPluginAdmin"]->getRawPluginDataFor($this->plugin_name);
+
+        if (!is_null($plugin) && $plugin['activation_possible'] && $plugin['supports_cli_setup']) {
+            $pl = ilPlugin::getPluginObject(
+                $plugin['component_type'],
+                $plugin['component_name'],
+                $plugin['slot_id'],
+                $plugin['name']
+            );
+
+            $pl->activate();
+        }
+
+        $GLOBALS["DIC"] = $ORIG_DIC;
+
+        return $environment;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function isApplicable(Setup\Environment $environment) : bool
+    {
+        $ORIG_DIC = $this->initEnvironment($environment);
```

Comment:
here i was confused at first because i did not expect the original DIC to be returned. to name the method correctly would be a bit cumbersome like `initTemporaryEnvironmentAndPreserveOrdiginal`... Let's leave it like this ;-)